src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+    Vec3f sp5C;
+    Vec3f sp50;
+    Camera* sp4C;
+    Player* player;
+    Camera* temp_v0;
+    Camera* temp_v0_2;
+    s32 pad;
+    f32 temp_f12;
+    f32 temp_f2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900890.s")
+    sp4C = Gameplay_GetCamera(globalCtx, 0);
+    player = PLAYER;
+    SkelAnime_Update(&this->skelAnime);
+    this->unk_398++;
+    this->unk_339 = 0x14;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80901020.s")
+    switch (this->unk_39C) {
+        case 0:
+            func_80064520(globalCtx, &globalCtx->csCtx);
+            this->unk_39E = Gameplay_CreateSubCamera(globalCtx);
+            Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+            Gameplay_ChangeCameraStatus(globalCtx, this->unk_39E, 7);
+            func_8002DF54(globalCtx, &this->actor, 8);
+            this->unk_39C = 1;
+            this->unk_3A4 = sp4C->eye;
+            this->unk_3B0 = sp4C->at;
+            this->unk_1A2[0] = 0x12C;
+            this->unk_1A2[1] = 0x64;
+            globalCtx->envCtx.unk_D8 = 0.0f;
+        case 1:
+            if (this->unk_1A2[1] == 0x32) {
+                func_80078884(NA_SE_EN_MGANON_WALK);
+            }
+            Matrix_RotateY(((this->actor.shape.rot.y / (f32)0x8000) * M_PI) + 0.3f, MTXMODE_NEW);
+            sp5C.x = 0.0f;
+            sp5C.y = 0.0f;
+            sp5C.z = 250.0f;
+            Matrix_MultVec3f(&sp5C, &sp50);
+            Math_ApproachF(&this->unk_3A4.x, this->actor.world.pos.x + sp50.x, 0.2f, 100.0f);
+            Math_ApproachF(&this->unk_3A4.y, 1136.0f, 0.2f, 100.0f);
+            Math_ApproachF(&this->unk_3A4.z, this->actor.world.pos.z + sp50.z, 0.2f, 100.0f);
+            Math_ApproachF(&this->unk_3B0.x, this->unk_1B8.x, 0.2f, 100.0f);
+            Math_ApproachF(&this->unk_3B0.y, this->unk_1B8.y, 0.2f, 100.0f);
+            Math_ApproachF(&this->unk_3B0.z, this->unk_1B8.z, 0.2f, 100.0f);
+            if (this->unk_1A2[1] == 0) {
+                this->unk_39C = 2;
+                this->unk_1A2[1] = 0x5A;
+            }
+            break;
+        case 2:
+            this->unk_1A2[0] = 0x12C;
+            this->unk_3A4.x = sZelda->actor.world.pos.x - 100.0f;
+            this->unk_3A4.y = sZelda->actor.world.pos.y + 30.0f;
+            this->unk_3A4.z = (sZelda->actor.world.pos.z + 30.0f) - 60.0f;
+            this->unk_3B0.x = sZelda->actor.world.pos.x;
+            this->unk_3B0.y = sZelda->actor.world.pos.y + 30.0f;
+            this->unk_3B0.z = sZelda->actor.world.pos.z - 10.0f;
+            Math_ApproachZeroF(&this->unk_324, 1.0f, 5.0f);
+            Math_ApproachF(&globalCtx->envCtx.unk_D8, 1.0f, 1.0f, 0.019607844f);
+            if (this->unk_1A2[1] == 0x50) {
+                func_8010B680(globalCtx, 0x70D7, NULL);
+            }
+            if ((this->unk_1A2[1] < 0x1E) && (func_8010BDBC(&globalCtx->msgCtx) == 0)) {
+                temp_v0 = Gameplay_GetCamera(globalCtx, 0);
```

Comment:
```suggestion
                temp_v0 = Gameplay_GetCamera(globalCtx, MAIN_CAM);
```
also in other places